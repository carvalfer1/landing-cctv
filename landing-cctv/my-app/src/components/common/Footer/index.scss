@import '../../../design/colors.scss';

.footer{
    border-top: 1px solid #f0f0f0;
    height: 200px;
    background-color: #f0f0f0;
    width: 100%;
    bottom: 0;
}