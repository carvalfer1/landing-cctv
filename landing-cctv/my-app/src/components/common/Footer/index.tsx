import * as React from 'react';
import './index.scss';

class Footer extends React.Component<{},{}>{
    
    render : any = () => {
        return <div className="footer"></div>
    }
}
export default Footer;