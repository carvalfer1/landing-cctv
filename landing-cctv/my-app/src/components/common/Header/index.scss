@import '../../../design/colors.scss';
.header{
    
}