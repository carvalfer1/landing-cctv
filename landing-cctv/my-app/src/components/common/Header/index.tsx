import * as React from 'react';
import { Link } from "react-router-dom";
import {Icon, Menu} from 'antd';
import './index.scss';
const { SubMenu } = Menu;

interface HeaderStates {
    current:string
}

class Header extends React.Component<{},HeaderStates>
{
    state = {
        current:""
    }    

    handleClick = (e:any) => {
        console.log(e.key);
        this.setState({current:e.key});
    }

    render : any = () => {
        return  <Menu className="header" onClick={this.handleClick} selectedKeys={[this.state.current]} mode="horizontal">
        <Menu.Item key="home">
          <Link to="/"><Icon type="home" />Home</Link>
        </Menu.Item>
        <SubMenu
          title={
            <span className="submenu-title-wrapper">
              <Icon type="setting" />
              Jest
            </span>
          }
        >
            <Menu.Item key="setting:1">
              <Link to="/jest" className="nav-text">Introduction</Link>
            </Menu.Item>
            <Menu.Item key="setting:2">Components mount</Menu.Item>
            <Menu.Item key="setting:3">Components functionality</Menu.Item>
         </SubMenu>
        <Menu.Item key="alipay">
          <a href="https://ant.design" target="_blank" rel="noopener noreferrer">
            Navigation Four - Link
          </a>
        </Menu.Item>
        <Menu.Item key="history">
          <Link to="/history"><Icon type="align-center" />Vote history</Link>
        </Menu.Item>
        <Menu.Item key="configuration">
          <Link to="/"><Icon type="align-center" />Configuration</Link>
        </Menu.Item>
      </Menu>;
    }
}

export default Header;