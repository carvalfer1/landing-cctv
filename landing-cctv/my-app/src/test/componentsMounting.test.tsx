import * as React from 'react';
import {doTestMount} from './ini';
import Header from '../components/common/Header';
import Footer from '../components/common/Footer';

doTestMount('Header Comp', <Header/>);
doTestMount('Footer Comp', <Footer/>);
