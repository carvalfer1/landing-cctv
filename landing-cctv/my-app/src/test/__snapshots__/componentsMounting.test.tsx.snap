// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Comp 1`] = `
<div
  className="footer"
/>
`;

exports[`Header Comp 1`] = `
<Menu
  className="header"
  mode="horizontal"
  onClick={[Function]}
  selectedKeys={
    Array [
      "",
    ]
  }
>
  <MenuItem
    key="mail"
  >
    <Icon
      type="mail"
    />
    Navigation One
  </MenuItem>
  <MenuItem
    disabled={true}
    key="app"
  >
    <Icon
      type="appstore"
    />
    Navigation Two
  </MenuItem>
  <SubMenu
    title={
      <span
        className="submenu-title-wrapper"
      >
        <Icon
          type="setting"
        />
        Navigation Three - Submenu
      </span>
    }
  >
    <MenuItemGroup
      disabled={true}
      title="Item 1"
    >
      <MenuItem
        key="setting:1"
      >
        Option 1
      </MenuItem>
      <MenuItem
        key="setting:2"
      >
        Option 2
      </MenuItem>
    </MenuItemGroup>
    <MenuItemGroup
      disabled={true}
      title="Item 2"
    >
      <MenuItem
        key="setting:3"
      >
        Option 3
      </MenuItem>
      <MenuItem
        key="setting:4"
      >
        Option 4
      </MenuItem>
    </MenuItemGroup>
  </SubMenu>
  <MenuItem
    key="alipay"
  >
    <a
      href="https://ant.design"
      rel="noopener noreferrer"
      target="_blank"
    >
      Navigation Four - Link
    </a>
  </MenuItem>
</Menu>
`;
